test_suite/README.md

```diff
@@ -25,172 +26,157 @@ These tests are used to validate ports of the API.
-* The commands executed should be displayed in the terminal. 
-* At the end of the test suite execution a summary of the results is displayed 
-  in the console.
+## List existing tests
 
+To list tests in the test suite:
 
-## Run specific tests 
+```
+pytest --collect-only
+```
 
-Pytest allows you to run tests in a specific folder or file or individual tests:
-- folder: `python -m pytest <folder name>`
-- test file: `python -m pytest <file path>` 
-- test within a file: `python -m pytest <file path>::<test name>`  
+## Run all tests
 
+1. Compile the `ble-cliapp` application for the boards you're using for testing.
+1. Flash the application to the boards.
+1. Run the command `pytest`.<!--where? In the same location as the mbed compile command?-->
 
-Multiple tests cases, test files or test path can be specified. They are separated 
-by spaces.
+    The terminal displays the executed commands.
+1. When all tests have run, the console displays a summary of the results.
 
-```
-python -m pytest <test folder1> <test folder 2> <file path> <file path>::<test name>
-```
+## Run specific tests
+
+pytest allows you to run tests in a specific folder or file or individual tests:
 
-## List existing tests 
+* Folder: `pytest <folder name>`
+* Test file: `pytest <file path>`
+* Test within a file: `pytest <file path>::<test name>`
 
-To list tests in the test suite use the command: 
+To use multiple tests cases, test files or test paths, separate them with spaces:
 
 ```
-python -m pytest --collect-only
+pytest <test folder1> <test folder 2> <file path> <file path>::<test name>
 ```
 
-## Run test with markers
+## Run tests by markers
+<!--"run with" can imply the marker changes the test; "run by" implies selecting tests that have the marker, which I think is the case here-->
 
-Some of the tests has been annotated with a pytest marker. Four markers are 
-available: 
-- ble41: Test of Bluetooth 4.1 features
-- ble42: Test of Bluetooth 4.2 features
-- ble50: Test of Bluetooth 5.0 features
-- smoketest: Small set of smoke tests.
+Some of the tests has been annotated with a pytest marker:
 
-Run the command `python -m pytest -m <marker name>` to run the test anotated with `marker name`.
+- `ble41`: Test of Bluetooth 4.1 features.
+- `ble42`: Test of Bluetooth 4.2 features.
+- `ble50`: Test of Bluetooth 5.0 features.
+- `smoketest`: Small set of smoke tests.
 
-## Select targets type
+To run the test annotated with `marker name`, run the command `pytest -m <marker name>`.
 
-To select specific target types that should run the test, use the command line 
-argument `--platforms=`. Multiple platforms can be specified, they must be separated 
-with a `,`. 
+## Select targets
 
-For example, this command runs tests on `NRF52840_DK`: 
+To select specific targets for the test, use the command line
+argument `--platforms=`. To specify multiple platforms, separate them with a comma `,`.<!--this isn't a type - it's a specific board, isn't it?-->
```

Comment:
It is not a specific board, but it is a specific target that should be used to run the tests. If two NRF52840_DK and two `DISCO_L475` are connected to the system we don't get to choose which one will be used to run the test but we can select the target used `NRF52840_DK` or `DISCO_L475`.

Reply:
But again, that's not a type - you can't select all NRF or all disco.

Reply:
if you pass the argument  `--platforms=NRF52840_DK` the test suite will consider all `NRF52840_DK` connected to your machine to run the test.

Reply:
So you can have multiples of the same board, but you can't just say "do all NRF boards"